internal/events/flagged_driver.go

```diff
@@ -0,0 +1,162 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+import (
+	"context"
+	"fmt"
+	"runtime/debug"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/open-feature/go-sdk/openfeature"
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/mindersec/minder/internal/engine/engcontext"
+	"github.com/mindersec/minder/internal/events/common"
+	"github.com/mindersec/minder/internal/flags"
+	serverconfig "github.com/mindersec/minder/pkg/config/server"
+)
+
+type flaggedDriver struct {
+	basePub       message.Publisher
+	experimentPub message.Publisher
+	baseSub       message.Subscriber
+	experimentSub message.Subscriber
+
+	flags             openfeature.IClient
+	publishedMessages metric.Int64Counter
+	readMessages      metric.Int64Counter
+}
+
+// Publish implements message.Publisher.  If the message is flagged by the
+// alternate_message_driver experiment, it is published to the experiment
+// driver, otherwise it is published to the base driver.
+func (f *flaggedDriver) Publish(topic string, messages ...*message.Message) error {
+	// Each message has its own context, so they _could_ be in different flag treatments
+	for _, m := range messages {
+		emptyContext := engcontext.EntityContext{}
```

Comment:
`emptyContext` can be moved outside the loop.

Reply:
Sounds good, though the cost is probably ~zero, since it is stack-allocated as the zero value.  (This is one of the times I wish Golang had `const`)

Reply:
Yeah, just a nit.